mountpoint-s3-client/src/s3_crt_client/get_object.rs

```diff
@@ -109,16 +145,32 @@ pub struct S3GetObjectRequest {
     finish_receiver: UnboundedReceiver<Result<GetBodyPart, Error>>,
     finished: bool,
     enable_backpressure: bool,
+    lazy_object_metadata: Lazy<Result<ObjectMetadata, Canceled>, Receiver<ObjectMetadata>>,
+    initial_read_window_empty: bool,
```

Comment:
Couldn't we use `next_offset == read_window_end_offset` instead?

Reply:
No. We'll risk to return an error when headers are available. We know that for any successful request with non empty  `initial_read_window` headers will eventually be available. 

The proposed `next_offset == read_window_end_offset` may not work when the `initial_read_window` is not empty and the request was read till the end. In this scenario the specified condition will evaluate to true. If `body_callback` is called before the `headers_callback` with a slight delay and a call to `get_object_metadata` happening between them an error will be returned.